numba/_dispatcher.cpp

```diff
@@ -115,28 +117,76 @@ typedef struct DispatcherObject{
     PyObject *argnames;
     /* Tuple of default values */
     PyObject *defargs;
-} DispatcherObject;
+    /* Number of arguments to function */
+    int argct;
+    /* Used for selecting overloaded function implementations */
+    TypeManager *tm;
+    /* An array of overloads */
+    Functions functions;
+    /* A flattened array of argument types to all overloads
+     * (invariant: sizeof(overloads) == argct * sizeof(functions)) */
+    TypeTable overloads;
```

Comment:
Is it intended for these to now be public?

Reply:
The have to be, because mixing public/private members results in a non-standard-layout class, so `offsetof` won't work with it then (used for implementing the members at https://github.com/numba/numba/pull/6349/files#diff-4374dcde8d4424f3951a6810c497d65972bfe81587b15ec6f91015a8b9dcd751R652).

Reply:
ahhh of course, `PyObject_HEAD` etc, and this is class not struct, could make it struct?

Reply:
It's possible to, I think it's just a stylistic thing. I prefer "class with public members" to "struct" in this case because I tend to expect classes to have methods and structs not to, but could change it if there's a compelling reason to go the other way.

Reply:
I'm ok with it either way, it's just the `PyObject_HEAD` making it more struct-like for `offsetof`, but agree RE methods. Will go with leaving as-is until something sufficiently compelling arrives to warrant change.